### DISCUSSION TITLE
Initial Qt Support

### DISCUSSION THREAD

**Discussion by jgilje** [86Box/86Box#1871] [Show and tell]
Hi all. Over at https://github.com/jgilje/86Box (qt branch), I've added Qt support to 86Box, primarily because I wanted a way to configure the VMs similar to what I had with PCem on Linux.

I've primarily worked on getting the Settings dialog ported from the Win32 code. The application output is split between a status/config window and a renderer window (similar to what PCem used to do), because I was unable to get the SDL window embedded into Qt. SDL_CreateWindowFrom never rendered anything, and while I got the renderer output embedded in Qt using Qt's QWindow + QWidget's createWindowContainer() - I never got this setup reliable regarding mouse grabbing and fullscreen support.

If its of any interest to get this into 86Box proper, let me know. If this is uninteresting for you - no worries :) I'll just keep a port over at my github page.

**Comment by OBattler** [comment]
Of course it's of interest! It would finally give us a proper UI on Linux.

**Comment by dob205** [comment]
For some reason the atomic structs seem to crash and burn the compile since they missed curly brackets, also you missed some adjustments in the `src/CMakeLists.txt` that broke the macOS bundle process, I'll add the fixed one into the attachments
[CMakeLists.txt](https://github.com/86Box/86Box/files/7602209/CMakeLists.txt). If you want direct feedback you can also join our Discord, which is linked in the `README` of our project.

**Reply by dhrdlicka** [reply]
Minor things such as mac bundles will be sorted out as time goes, let's get the thing finished on Linux in the first place.

**Reply by dob205** [reply]
I mean it doesn't hurt since the bundle stuff is rather easy to add and doesn't detract the goal of getting it finished on Linux.

**Reply by jgilje** [reply]
yeah, i'll get it added! joined the discord btw :)

**Reply by dob205** [reply]
Also thank you very much for the Qt port! Wonder how you got the new icons loaded.

**Reply by jgilje** [reply]
Nothing special done be me, so I guess Qt simply has support for them?  Is there anything special about them, other than just being .ico files?

**Reply by dob205** [reply]
That's actually great to hear!
